ENIGMAsystem/SHELL/Platforms/General/UNIXfilemanip.cpp

```diff
@@ -20,48 +20,626 @@
 #include <unistd.h>
 #include <sys/stat.h>
 
+#include <stdio.h>
+#include <libgen.h>
+#include <dirent.h>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+
+#include <fts.h>
+#include <stdlib.h>
+#include <limits.h>
+
 #include <string>
 #include "PFfilemanip.h"
 using namespace std;
 
 /* UNIX-ready port of file manipulation */
 
+struct dupslash
```

Comment:
This is a nice gesture, but it's unnecessary. The filesystem doesn't care if there are duplicate slashes (and your code won't mind, either).

Reply:
I added that mainly to prevent directory_copy from infinitely copying inside itself.

Later on in the code you will see I do a substring comparison and if argument0 of directory_copy is a substring of argument1, then prevent the copy, otherwise it will infinitely copy itself inside itself with infinite subdirectories until you machince runs out of space. The two strings need to match, that's why I removed the duplicate slashes and used realpath(). I still have dupslash in there in case if realpath() fails, because yes I know realpath also removes extra slashes.

...or is that not an issue I should worry about catering to?

Reply:
Anyways let me know and I'll change it.

Reply:
Oh, I also did it for file_copy to make sure it returns 0 if argument0 = argument1, in which case will also need either dupslash, realpath, or both, (afaik, at least)